traffic_portal/test/end_to_end/users/users-spec.js

```diff
@@ -41,6 +57,84 @@ describe('Traffic Portal Users Test Suite', function() {
+	});
+
+	it('should create a new user', function () {
+		console.log("Creating a new user");
+		expect(pageData.createButton.isEnabled()).toBe(false);
+		pageData.username.sendKeys(myNewUser.username);
+		pageData.fullName.sendKeys(myNewUser.fullName);
+		pageData.email.sendKeys(myNewUser.email);
+		pageData.roleName.sendKeys(myNewUser.roleName);
+		pageData.tenantId.sendKeys(myNewUser.tenantId);
+		pageData.localPasswd.sendKeys(myNewUser.localPasswd);
+		pageData.confirmLocalPasswd.sendKeys(myNewUser.confirmLocalPasswd);
+		expect(pageData.createButton.isEnabled()).toBe(true);
+		pageData.createButton.click();
+		expect(browser.getCurrentUrl().then(commonFunctions.urlPath)).toEqual(commonFunctions.urlPath(browser.baseUrl)+"#!/users");
+	});
+
+	it('should update a user', function() {
+		console.log('Updating the new user: ' + myNewUser.username);
+		browser.sleep(250);
+		pageData.searchFilter.sendKeys(myNewUser.username);
+		browser.sleep(250);
+		element.all(by.repeater('u in ::users')).filter(function(row){
+			return row.element(by.name('username')).getText().then(function(val){
+				return val === myNewUser.username;
+			});
+		}).get(0).click();
+		browser.sleep(1000);
+		pageData.fullName.clear();
+		pageData.fullName.sendKeys(myNewUser.fullName + ' updated');
+		pageData.updateButton.click();
+		expect(pageData.fullName.getText() === myNewUser.fullName + ' updated');
+	});
+
+	it('should open new registered users form page', function() {
+		console.log("Open new register users form page");
+		browser.setLocation("users");
+		pageData.registerNewUserButton.click();
+		expect(browser.getCurrentUrl().then(commonFunctions.urlPath)).toEqual(commonFunctions.urlPath(browser.baseUrl)+"#!/users/register");
+	});
+
+	it('should create a new registered user', function () {
+		console.log("Creating a new registered user");
+		expect(pageData.registerEmailButton.isEnabled()).toBe(false);
+		pageData.email.sendKeys(myNewRegisteredUser.email);
+		pageData.roleName.sendKeys(myNewRegisteredUser.roleName);
+		pageData.tenantId.sendKeys(myNewRegisteredUser.tenantId);
+		expect(pageData.registerEmailButton.isEnabled()).toBe(true);
+		pageData.registerEmailButton.click();
+		expect(browser.getCurrentUrl().then(commonFunctions.urlPath)).toEqual(commonFunctions.urlPath(browser.baseUrl)+"#!/users/register");
+	});
+
+	it('should update a new registered user', function() {
+		console.log('Updating the new registered user: ' + myNewRegisteredUser.email);
+		browser.setLocation("users");
+		pageData.searchFilter.clear();
+		browser.sleep(250);
+		pageData.searchFilter.sendKeys(myNewRegisteredUser.email);
+		browser.sleep(250);
+		element.all(by.repeater('u in ::users')).filter(function(row){
+			return row.element(by.name('email')).getText().then(function(val){
+				return val === myNewRegisteredUser.email;
+			});
+		}).get(0).click();
+		browser.sleep(1000);
+		expect(pageData.registerSent.isEnabled()).toBe(false);
```

Comment:
line 132 and 133 seem redundant and probably won't work with the `readonly` attribute

Reply:
I think I changed it before your comment. So line 132 is not present.